### PR TITLE
Add support for targeting cross_attn layers in mllama

### PR DESCRIPTION
**Pull request by ajtejankar**
Previous PR that added support for mllama models didn't allow targeting cross attention layers in the text model. This PR adds support for it. I think the code is ready, but it needs to be tested with an adapter.

### CODE REVIEW

**Comment by ajtejankar** [comment]
I've verified this against an adapter targeting all modules that was trained on a small subset of the [Doc VQA dataset](https://huggingface.co/datasets/cmarkea/doc-vqa). It's ready to merge.

**Review by arnavgarg1 — APPROVED** [review]
LGTM! Let's merge after resolving the merge conflict

**Pull request merged**
